### ISSUE TITLE
Remove device - Directed to Operational Error webpage

### ISSUE THREAD

**Issue by Mikmonken** [thorrak/fermentrack#125]
Django Admin page directs to operational error, same issue also occurs if the logging in stopped.

effectively can't remove the device without getting directed to this issue.

```
Environment:


Request Method: POST
Request URL: http://fermentrack.local/admin/app/brewpidevice/

Django Version: 1.10.6
Python Version: 2.7.9
Installed Applications:
['suit',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'app.apps.AppConfig',
 'constance',
 'constance.backends.database']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback:

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  42.             response = get_response(request)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/options.py" in wrapper
  544.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  57.         response = view_func(request, *args, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/sites.py" in inner
  211.             return view(request, *args, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/utils/decorators.py" in _wrapper
  67.             return bound_func(*args, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/utils/decorators.py" in bound_func
  63.                 return func.__get__(self, type(self))(*args2, **kwargs2)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/options.py" in changelist_view
  1569.                 response = self.response_action(request, queryset=cl.get_queryset(request))

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/options.py" in response_action
  1305.             response = func(self, request, queryset)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/actions.py" in delete_selected
  37.         queryset, opts, request.user, modeladmin.admin_site, using)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/utils.py" in get_deleted_objects
  128.     collector.collect(objs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/utils.py" in collect
  193.             return super(NestedObjects, self).collect(objs, source_attr=source_attr, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/deletion.py" in collect
  227.                         field.remote_field.on_delete(self, field, sub_objs, self.using)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/deletion.py" in CASCADE
  17.                       source_attr=field.name, nullable=field.null)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/contrib/admin/utils.py" in collect
  193.             return super(NestedObjects, self).collect(objs, source_attr=source_attr, **kwargs)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/deletion.py" in collect
  226.                     elif sub_objs:

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/query.py" in __nonzero__
  264.         return type(self).__bool__(self)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/query.py" in __bool__
  260.         self._fetch_all()

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/query.py" in _fetch_all
  1087.             self._result_cache = list(self.iterator())

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/query.py" in __iter__
  54.         results = compiler.execute_sql()

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/models/sql/compiler.py" in execute_sql
  835.             cursor.execute(sql, params)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/backends/utils.py" in execute
  79.             return super(CursorDebugWrapper, self).execute(sql, params)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/backends/utils.py" in execute
  64.                 return self.cursor.execute(sql, params)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/utils.py" in __exit__
  94.                 six.reraise(dj_exc_type, dj_exc_value, traceback)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/backends/utils.py" in execute
  64.                 return self.cursor.execute(sql, params)

File "/home/fermentrack/venv/lib/python2.7/site-packages/django/db/backends/sqlite3/base.py" in execute
  337.         return Database.Cursor.execute(self, query, params)

Exception Type: OperationalError at /admin/app/brewpidevice/
Exception Value: no such table: app_beerlogpoint
```

**Comment by thorrak** [comment]
Interesting - I hadn't thought about this one. I overrode the save function in the BeerrLogPoint model, I hadn't thought about needing to override the delete function. Should be a quick fix when I'm home later tonight.

**Comment by Mikmonken** [comment]
Just as an update I still can't delete my Test ESP it doesn't seem to be logging anything at the moment either

**Comment by thorrak** [comment]
I've got a fix for this in the `fixes` branch. I'm going to try to fix #126 as well (probably tomorrow) and will then merge both tomorrow evening.
